### PR TITLE
Overhaul contract deployment and configuration for local demo

### LINKED ISSUE #649: Problem: Deployment involves too many manual steps
There are currently quite a few steps in deploying our cloud demo. Roughly:

1. Deploy hotshot contract -> deployment artifact
2. Start sequencer
3. Deploy L2 contracts -> deployment artifact
4. Start L2 services

In addition deployment outputs for the contract deployments need to be manually handled to make them inputs to subsequent tasks.

Step 1. and 3. require a funded wallet and infura and etherscan account.

I think we can do better feel free to leave ideas below. 

- A simple improvement would be to store the L2 contracts deployment artifacts inside the repo (similar to what `hardhat-deploy` and `forge script` are doing).

cc @Ancient123 @nomaxg @ImJeremyHe @jbearer

### PR DESCRIPTION
**Pull request by jbearer**
* New rust script to deploy contracts
* Run script in docker and native demos before starting commit and prover services
* Remove deployment options from commit and prover services

Closes #649

### CODE REVIEW

File: sequencer/src/bin/deploy.rs [hunk 1]
```diff
@@ -0,0 +1,332 @@
...
+        let file = File::options().create(true).write(true).open(out)?;
+        contracts.write(file)?;
+    } else {
+        contracts.write(stdout())?;
+    }
+
+    Ok(())
+}
+
+async fn deploy_light_client_contract<M: Middleware + 'static>(
```
[the remark this answers is not in the document]
Comment (mrain) [reply sequencer/src/bin/deploy.rs:280]: This looks awesome!
We could replace `gen-demo-genesis` by this deployment fn.

File: docker/prover-service.Dockerfile [hunk 2]
```diff
@@ -24,15 +24,11 @@ COPY contracts/ /work/contracts/
```
Comment (mrain) [inline docker/prover-service.Dockerfile:24]: I think rust and foundry is no longer needed in this docker.

File: docker/prover-service.Dockerfile [hunk 3]
```diff
@@ -24,15 +24,11 @@ COPY contracts/ /work/contracts/
...
-RUN chmod +x /usr/local/bin/launch-prover-service
-RUN chmod +x /usr/local/bin/gen-demo-genesis
 
 # When running as a Docker service, we always want a healthcheck endpoint, so set a default for the
 # port that the HTTP server will run on. This can be overridden in any given deployment environment.
 ENV ESPRESSO_COMMITMENT_TASK_PORT=80
 HEALTHCHECK --interval=1s --timeout=1s --retries=100 CMD curl --fail http://localhost:${ESPRESSO_COMMITMENT_TASK_PORT}/healthcheck || exit 1
 
-CMD [ "launch-prover-service" ]
+CMD [ "state-prover", "-d" ]
```
Comment (mrain) [inline docker/prover-service.Dockerfile:34]: Rust and foundry is no longer needed in this docker.

Comment (jbearer) [reply docker/prover-service.Dockerfile:34]: Good point

File: sequencer/src/bin/deploy.rs (hunk 1 above)
[the remark this answers is not in the document]
Comment (mrain) [reply sequencer/src/bin/deploy.rs:280]: Second thought, we could still keep `gen-demo-genesis`.

Do you think it's a good idea to separate light client contract deployment from the hotshot contract deployment?
Some user may not need the light client.

File: scripts/build-docker-images [hunk 4]
```diff
@@ -41,7 +41,6 @@ done
 cp -r "${CONTRACTS_DIR}" ${WORKDIR}
```
Comment (mrain) [inline scripts/build-docker-images:41]: Do we still need this?

File: scripts/build-docker-images [hunk 5]
```diff
@@ -41,7 +41,6 @@ done
...
 cp -v "./foundry.toml" ${WORKDIR}
 mkdir -p ${WORKDIR}/scripts
```
[the remark this answers is not in the document]
Comment (jbearer) [reply scripts/build-docker-images]: Removed in [6d5f487](https://github.com/EspressoSystems/espresso-sequencer/pull/1262/commits/6d5f48768aa02fbfa2523e3c8c81719dc5549961)

File: scripts/build-docker-images (hunk 4 above)
Comment (jbearer) [reply scripts/build-docker-images]: Removed in [6d5f487](https://github.com/EspressoSystems/espresso-sequencer/pull/1262/commits/6d5f48768aa02fbfa2523e3c8c81719dc5549961)

File: docker/prover-service.Dockerfile (hunk 3 above)
Comment (jbearer) [reply docker/prover-service.Dockerfile:19]: Removed in [6d5f487](https://github.com/EspressoSystems/espresso-sequencer/pull/1262/commits/6d5f48768aa02fbfa2523e3c8c81719dc5549961)

File: sequencer/src/bin/deploy.rs (hunk 1 above)
[the remark this answers is not in the document]
Comment (jbearer) [reply sequencer/src/bin/deploy.rs:280]: The point of this script is to set the L1 up for demo or staging. So I think in this case it makes sense to have them together. But it is also beneficial to be able to deploy with foundry so I think it's a good idea to keep `gen-demo-gensis`

**Review by mrain — APPROVED** [review]
Awesome work. Thanks!!

File: sequencer/src/bin/deploy.rs
Comment (alxiong) [inline sequencer/src/bin/deploy.rs:1]: thanks for the great work! @jbearer these APIs looks very easy to use and thanks for solving the half-baked solution earlier and simplifying our deployment code! 🚀

**Pull request merged**